### DISCUSSION TITLE
automation/guides/pihole-doh/

### DISCUSSION THREAD

**Discussion by giscus** [fire1ce/3os.org#73] [General]
# automation/guides/pihole-doh/

Pi-hole as DNS Server with DNS over HTTPS (DOH) Based on Docker Containers. Pi-hole, DNS ads, tracking blocking on Ubuntu with DNS over HTTP, list of blacklist

https://3os.org/automation/guides/pihole-doh/

**Comment by amokster88** [comment]
Great Guide Thanks so much! I was finding everywhere for a good docker compose stack. At last I found it. Thank you very much! It works great!

**Reply by fire1ce** [reply]
You are more than welcome. Enjoy!

**Reply by amokster88** [reply]
Please also do a compose stack with DOT(DNS over TLS) too

**Reply by fire1ce** [reply]
You better stick with DOH than DOT. 
And since i don't use DOT i won't post something i've didn't test.

**Reply by amokster88** [reply]
Yeah, DoT and DoH use the same TLS encryption layer with the same PKI, each have their own advantage and disadvantage but I thought it would be nice to have options 😜 
Well, thanks have a good day!

**Comment by LuisMLGDev** [comment]
Thanks for the guide!!
I wonder why use the same URL for both UPSTREAMs?
